### PR TITLE
Performance: Preload media REST API request in editor

### LINKED ISSUE #3197: Editor: Preload media REST API request
## Feature Description

Similar to what was done in #2933, we can leverage the same `_web_stories_envelope` technique to markedly improve perceived performance when opening the editor by preloading requests to the Media REST API endpoint.

## Additional Context

Eventually this will be fixed in `@wordpress/api-fetch` so that we don't have to maintain this `_web_stories_envelope` logic ourselves anymore. See https://github.com/WordPress/gutenberg/pull/23807

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

* When opening the editor, now extra HTTP request is made for the Media Library, as the request is **preloaded** upon page load.

## Implementation Brief

* Basically adapt #2933 for media requests.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Preload media on the editor to stop the loading message from being displayed. 

## Testing Instructions

---

Fixes #3197

### CODE REVIEW

**Review by swissspidy — COMMENTED** [review]
Looks good but needs to be brought up to date with `main` and some lint fixes

**Review by swissspidy — CHANGES_REQUESTED** [review]
Needs lint fixes

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
Not sure why, but the `MediaPane fetching should fetch 2nd page` karma test fails with this PR

File: assets/src/dashboard/dataUtils/formattedStoriesArray.js [hunk 1]
```diff
@@ -249,7 +249,8 @@ const formattedStoriesArray = [
         ],
         'wp:attachment': [
           {
-            href: 'http://localhost:8899/wp-json/wp/v2/media?parent=167',
+            href:
+              'http://localhost:8899/wp-json/web-stories/v1/media?parent=167',
```
Comment (diegovar) [inline assets/src/dashboard/dataUtils/formattedStoriesArray.js:253]: Can you explain why this change is needed?

File: assets/src/edit-story/app/api/apiProvider.js [hunk 2]
```diff
@@ -113,6 +113,7 @@ function APIProvider({ children }) {
         context: 'edit',
         per_page: perPage,
         page: pagingNum,
+        _web_stories_envelope: 1,
```
Comment (diegovar) [inline assets/src/edit-story/app/api/apiProvider.js]: Why do you need to use 1 instead of true?

**Review by diegovar — COMMENTED** [review]
Can this be guarded with a feature flag?

File: assets/src/dashboard/dataUtils/formattedStoriesArray.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/dashboard/dataUtils/formattedStoriesArray.js:253]: @diegovar The REST API URL in WordPress changed from including `/wp/v2` to `/web-stories/v1`. This change here merely reflects that. The rest is Prettier.

File: assets/src/edit-story/app/api/apiProvider.js (hunk 2 above)
Comment (swissspidy) [reply assets/src/edit-story/app/api/apiProvider.js]: Both works, just needs to be truthy. Will change to `true` though for clarity.
